### PR TITLE
feat(searchbar): autocapitalize, dir, lang, maxlength, and minlength are inherited to native input

### LINKED ISSUE #27606: bug: additional properties for searchbar
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

The input inside `ion-searchbar` does not support all possible attributes. 

```html
<ion-searchbar animated="true" search-icon="code-working-outline" 
aria-label="Spoken Language Text" placeholder="Enter Text" autocomplete="off" autocorrect="off" spellcheck="false" 
autofocus dir="auto" aria-autocomplete="list" autocapitalize="off" data-gramm_editor="false" lang="en" maxlength="100">
```

Results in:
```html
<input 
aria-label="search text" placeholder="Enter Text" autocomplete="off" autocorrect="off" spellcheck="false">
```

### Expected Behavior

Should inherit attributes from `ion-searchbar`.

Input should also have 
```html
autofocus dir="auto" aria-autocomplete="list" autocapitalize="off" data-gramm_editor="false" lang="en" maxlength="100"
```

### Steps to Reproduce

Create an `ion-searchbar`:

```html
<ion-searchbar animated="true" search-icon="code-working-outline" 
aria-label="Spoken Language Text" placeholder="Enter Text" autocomplete="off" autocorrect="off" spellcheck="false" 
autofocus dir="auto" aria-autocomplete="list" autocapitalize="off" data-gramm_editor="false" lang="en" maxlength="100">
```

### Code Reproduction URL

_No response_

### Ionic Info

Ionic:

   Ionic CLI : 7.1.1

Utility:

   cordova-res : 0.15.4
   native-run  : 1.7.2

System:

   NodeJS : v18.15.0
   npm    : 9.6.7
   OS     : macOS Unknown

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: resolves #27606

---------

## What is the current behavior?

Certain attributes are not be inherited to the inner searchbar. Developers need control over these attributes to provide important context to users for things like language and text direction. Additionally, being able to control things like autocapitalize, maxlength, and minlength can help improve the user experience by a) guiding what should be entered into an input and b) removing autocapitalize where it's not appropriate. 

## What is the new behavior?

- Added autocapitalize, maxlength, and minlength properties
- lang and dir are global attributes, so adding them as properties will cause issues. However, developers can still set them as attributes and they will be inherited to the native `input` element. We also watch them so any changes to the attributes are also inherited to the native `input`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Note: We expanded the scope of this work to also include input and textarea, and this work will be handled separately. However, the original request was only for searchbar so that's why I associated this PR with the linked issue.

Dev build: `7.7.3-dev.11709159644.114cd8b1`

### CODE REVIEW

File: core/src/components/searchbar/test/searchbar.spec.ts
Comment (liamdebeasi) [inline core/src/components/searchbar/test/searchbar.spec.ts:6]: I did not test updating the properties because that would be testing Stencil behavior. I am, however, testing updating attributes to verify that I configured the watchers correctly.

File: core/src/components/searchbar/searchbar.tsx
Comment (liamdebeasi) [inline core/src/components/searchbar/searchbar.tsx:93]: I can make a tech debt ticket for this, but my plan was to make the change once this is merged since Ionic 8 is in beta now (and we can still make breaking changes).

**Review by amandaejohnston — APPROVED** [review]

**Comment by liamdebeasi** [comment]
Follow up PR: https://github.com/ionic-team/ionic-framework/pull/29107

Blocked until `feature-7.8` is merged into `main` (so it can be synced back to `feature-8.0`.

**Pull request merged**
